### PR TITLE
Avoid race condition in ArgumentsReflector.csproj

### PR DESCRIPTION
**Pull request by ViktorHofer**
ArgumentsReflector.csproj was built twice to the same output which caused the build to fail inside the VMR:
https://github.com/dotnet/installer/pull/18762

This was happening because ArgumentForwarding.Tests.csproj was building ArgumentsReflector.csproj as well but with an additional global property.

Instead, make the test app publish after build and copy its publish output into ArgumentForwarding.Tests.csproj' output/publish folder.

### CODE REVIEW

**Comment by ViktorHofer** [comment]
@ericstj can you please take a look? I'm quite sure that this can be simpler by setting an item that then automatically gets copied to the output and publish directory. But unfortunately I forgot that item's name :(

Including a ProjectReference in an app's (publish) output isn't natively supported by the SDK, right?

**Review by akoeplinger — APPROVED** [review]

**Comment by baronfel** [comment]
> @ericstj can you please take a look? I'm quite sure that this can be simpler by setting an item that then automatically gets copied to the output and publish directory. But unfortunately I forgot that item's name :(
> 
> Including a`ReferenceOutputAssembly=false` ProjectReference in an app's (publish) output isn't natively supported by the SDK, right?

I made a sample last week of how to include an projects' publish outputs in another projects' publish outputs: https://github.com/baronfel/msbuild-nested-app-sample

Maybe that helps? It has an Item type that I think would be useful, as well as a good extension point to do the addition at during the publish of the referencing app.

File: test/ArgumentForwarding.Tests/ArgumentForwarding.Tests.csproj [hunk 1]
```diff
@@ -10,22 +10,27 @@
   <ItemGroup>
     <ProjectReference Include="..\..\src\Cli\dotnet\dotnet.csproj" />
     <ProjectReference Include="..\Microsoft.NET.TestFramework\Microsoft.NET.TestFramework.csproj" />
+    <ProjectReference Include="..\ArgumentsReflector\ArgumentsReflector.csproj" ReferenceOutputAssembly="false" OutputItemType="ArgumentsReflectorOutput" />
```
Comment (baronfel) [inline test/ArgumentForwarding.Tests/ArgumentForwarding.Tests.csproj:13]: Consider setting the `Targets` metadata to `Publish` - this'll overwrite the default of `Build` and you won't have to modify `ArgumentsReflector.csproj` at all.

**Comment by ViktorHofer** [comment]
Thanks @baronfel and @KalleOlaviNiemitalo. I combined both of your ideas which achieves:
- Publishing the dependency without touching its project file
- Avoiding any hardcoded paths
- Using the CopyToOutputDirectory and CopyToPublishDirectory metadata to avoid the copy task

AFAIK the only downside with that approach is that VS doesn't support the `Target` metadata and the SDK doesn't like duplicate P2Ps inside VS.

**Comment by ericstj** [comment]
It looks to me like the only purpose of this test project `ArgumentForwarding.Tests` is to test the test framework itself and how it formats commandlines.

I see `reflector_cmd.cmd` is only ever used by the `ArgumentForwarding.Tests` -- so I'd suggest to just move that content item to the test project.

Then you could make the reference to `ArgumentsReflector.csproj` just a normal ProjectReference.  Looks to me like this project's only purpose is to echo back the parameters passed to it.  It doesn't need to be responsible for copying that cmd script - just it's deps/runtimeconfig.

**Review by ericstj — APPROVED** [review]
Looks good to me assuming this helper exe produces the the deps and runtimeconfig during build and the tests pass.  :shipit:

**Pull request merged**